File: DIO/20_Logica_programacao/Desafioobjeto_DIO.js
// Definição da classe Heroi
class Heroi {
    constructor(nome, idade, tipo) {
      this.nome = nome;
      this.idade = idade;
      this.tipo = tipo;
    }
  
    atacar() {
      let ataque;
      switch (this.tipo) {
        case 'mago':
          ataque = 'usou magia';
          break;
        case 'guerreiro':
          ataque = 'usou espada';
          break;
        case 'monge':
          ataque = 'usou artes marciais';
          break;
        case 'ninja':
          ataque = 'usou shuriken';
          break;
        default:
          ataque = 'usou um ataque';
      }
      console.log(`O ${this.tipo} atacou usando ${ataque}`);
    }
  }
  
  // Exemplo de uso da classe Heroi
  const heroi1 = new Heroi('Gandalf', 1000, 'mago');
  heroi1.atacar();
  
  const heroi2 = new Heroi('Aragorn', 35, 'guerreiro');
  heroi2.atacar();
  
  const heroi3 = new Heroi('Bruce Lee', 32, 'monge');
  heroi3.atacar();
  
  const heroi4 = new Heroi('Hanzo', 28, 'ninja');
  heroi4.atacar();
  

File: DIO/20_Logica_programacao/DesafioObjeto_DIO.c
#include <stdio.h>
#include <string.h>

// Definição da estrutura Heroi
typedef struct {
    char nome[50];
    int idade;
    char tipo[20];
} Heroi;

// Função para atacar
void atacar(Heroi *heroi) {
    char ataque[50];
    
    if (strcmp(heroi->tipo, "mago") == 0)
        strcpy(ataque, "usou magia");
    else if (strcmp(heroi->tipo, "guerreiro") == 0)
        strcpy(ataque, "usou espada");
    else if (strcmp(heroi->tipo, "monge") == 0)
        strcpy(ataque, "usou artes marciais");
    else if (strcmp(heroi->tipo, "ninja") == 0)
        strcpy(ataque, "usou shuriken");
    else
        strcpy(ataque, "usou um ataque");
    
    printf("O %s de nome %s, idade %d, atacou usando %s\n", heroi->tipo, heroi->nome, heroi->idade, ataque);
}

int main() {
    Heroi heroi;
    
    // Entrada de dados
    printf("Digite o nome do heroi: ");
    scanf("%s", heroi.nome);
    
    printf("Digite a idade do heroi: ");
    scanf("%d", &heroi.idade);
    
    printf("Escolha o tipo de heroi (mago, guerreiro, monge, ninja): ");
    scanf("%s", heroi.tipo);
    
    // Chama a função para atacar
    atacar(&heroi);

    return 0;
}